src/Commands/PowerPlatform/PowerApps/ExportPowerApp.cs

```diff
@@ -0,0 +1,181 @@
+
+        [Parameter(Mandatory = false)]
+        public string OutPath;
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter Force;
+
+        protected override void ExecuteCmdlet()
+        {
+            if (ParameterSpecified(nameof(OutPath)))
+            {
+                if (!System.IO.Path.IsPathRooted(OutPath))
+                {
+                    OutPath = System.IO.Path.Combine(SessionState.Path.CurrentFileSystemLocation.Path, OutPath);
+                }
+                if (System.IO.Directory.Exists(OutPath))
+                {
+                    throw new PSArgumentException("Please specify a folder including a filename");
+                }
+                if (System.IO.File.Exists(OutPath))
+                {
+                    if (!Force && !ShouldContinue($"File '{OutPath}' exists. Overwrite?", "Export App"))
+                    {
+                        // Exit cmdlet
+                        return;
+                    }
+                }
+            }
+
+            var environmentName = Environment.GetName();
+            var appName = Identity.GetName();
+
+            var postData = new
+            {
+                baseResourceIds = new[] {
+                    $"/providers/Microsoft.PowerApps/apps/{appName}"
+                }
+            };
+            var wrapper = RestHelper.PostAsync<Model.PowerPlatform.PowerApp.PowerAppPackageWrapper>(Connection.HttpClient, $"https://api.bap.microsoft.com/providers/Microsoft.BusinessAppPlatform/environments/{environmentName}/listPackageResources?api-version=2016-11-01", AccessToken, payload: postData).GetAwaiter().GetResult();
+
+            if (wrapper.Status == Model.PowerPlatform.PowerApp.Enums.PowerAppExportStatus.Succeeded)
+            { 
+                foreach (var resource in wrapper.Resources)
+                {
+                    if (resource.Value.Type == "Microsoft.PowerApps/apps")
+                    {
+                        resource.Value.SuggestedCreationType = "Update";
+                    }
+                }
+            var exportPostData = new
+            {
+                includedResourceIds = new[]
+                {
+                             $"/providers/Microsoft.PowerApps/apps/{appName}"
+                },
+                details = new
+                {
+                    displayName = PackageDisplayName,
+                    description = PackageDescription,
+                    creator = PackageCreatedBy,
+                    sourceEnvironment = PackageSourceEnvironment
+                },
+                resources = wrapper.Resources
+            };
+
+            var responseHeader = RestHelper.PostAsyncGetResponseHeader<string>(Connection.HttpClient, $"https://api.bap.microsoft.com/providers/Microsoft.BusinessAppPlatform/environments/{environmentName}/exportPackage?api-version=2016-11-01", AccessToken, payload: exportPostData).GetAwaiter().GetResult();
```

Comment:
@gautamdsheth  @KoenZomers 
To Export a PowerApp Package we need to make multiple API Calls, and this is 2nd call /exportpackage which returns us the location of another URL which will tell us when the download is ready but we have to keep of sending request.... and then when the status is successed it would return the url of zip file.  Now the location property from this API call comes in response Headers and not as a response content so I had to introduce new functions in RestHelper which will return us headers to read the response headers properties. 

So if you check in RestHelper, I have added two new functions...open to suggestions on the best way to handle this?